canonical/jaas-dashboard
# WD-2610 - Improve the error boundary

## Done

- Migrate ErrorBoundary to TypeScript.
- Display error and stack trace on error page.

## QA

- Pull code
- Run `dotrun clean && dotrun serve`
- Open http://0.0.0.0:8036/
- The easiest way to see the error boundary is to throw an error e.g add `throw new Error("It's definitely broken");` to ModelsIndex.tsx.
- Check that the error and stack trace are shown.
- Click the "raise an issue" link and check that the stack trace is included in the issue.

## Details

Fixes: #525.
https://warthogs.atlassian.net/browse/WD-2610

## Screenshots

<img width="976" alt="Screenshot 2023-03-14 at 12 41 12 pm" src="https://user-images.githubusercontent.com/361637/224870674-5770d5e2-7d67-4ebd-98b6-5e44a6bd303d.png">
<img width="1099" alt="Screenshot 2023-03-14 at 11 38 52 am" src="https://user-images.githubusercontent.com/361637/224870661-2863b5a2-8b79-4aab-b1c5-100dc8ae5c95.png">
